assembl/models/idea.py

```diff
@@ -999,11 +999,30 @@ class RootIdea(Idea):
 
     @property
     def num_posts(self):
-        """ In the root idea, this is the count of all mesages in the system """
+        """ In the root idea, num_posts is the count of all non-deleted mesages in the discussion """
+        from .post import Post
+        result = self.db.query(Post).filter(
+            Post.discussion_id == self.discussion_id,
+            Post.hidden==False,
+            Post.tombstone_condition()
+        ).count()
+        return int(result)
+
+    @property
+    def num_read_posts(self):
+        """ In the root idea, num_posts is the count of all non-deleted read mesages in the discussion """
         from .post import Post
```

Comment:
When trying to avoid circular import you usually move your imports to the end of the file: http://effbot.org/zone/import-confusion.htm (see circular imports). Or maybe I don't understand how you prevent circular import this way. In this case, I'd be interested to understand how you do that. Besides, in my point of view, using an import in the middle of a function will cause that function to take longer. And finally, all python style guides encourage to define imports at the beginning of a file. But discussion is opened on this topic.